UMS/man/man3/listSessions.3
'\" t
.\"     Title: listSessions
.\"    Author:  SysFera SAS <info@sysfera.com>
.\" Generator: DocBook XSL Stylesheets v1.75.2 <http://docbook.sf.net/>
.\"      Date: February 2011
.\"    Manual: C++ API Reference
.\"    Source: VISHNU 1.0
.\"  Language: English
.\"
.TH "LISTSESSIONS" "3" "February 2011" "VISHNU 1.0" "C++ API Reference"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
listSessions \- lists all sessions of the user
.SH "SYNOPSIS"
.HP \w'int\ vishnu::listSessions('u
.BI "int vishnu::listSessions(const\ string&\ " "sessionKey" ", ListSessions&\ " "listsession" ", const\ ListSessionOptions&\ " "options\ =\ ListSessionOptions()" ");"
.SH "DESCRIPTION"
.PP
This command is used to display and filter the list of all your sessions\&. For each session, a session identifier is provided which you can use to reconnect to a given session using the vishnu_reconnect command\&. The session\*(Aqs status is either 0 (inactive) or 1 (active)\&.
.SH "ARGUMENTS"
.PP
\fIsessionKey\fR
.RS 4
Input argument\&. The sessionKey is the encrypted identifier of the session generated by VISHNU\&.
.RE
.PP
\fIlistsession\fR
.RS 4
Output argument\&. Listsession is the list of sessions \&.
.RE
.PP
\fIoptions\fR
.RS 4
Input argument\&. Allows the user to list sessions using several optional criteria such as: the state of sessions (actives or inactives, by default, all sessions are listed), a period, a specific session or for admin to list all sessions of all users or sessions of a specific user\&.\&.
.RE
.SH "EXCEPTIONS"
.PP
The following exceptions may be thrown:
.PP
"Internal Error: SysDS service failure" [1]
.RS 4
.RE
.PP
"Internal Error: Database returned an error" [2]
.RS 4
.RE
.PP
"Internal Error: Database connection failure" [3]
.RS 4
.RE
.PP
"Internal Error: System call failure" [4]
.RS 4
.RE
.PP
"Internal Error: Undefined exception" [9]
.RS 4
.RE
.PP
"The userId is unknown" [21]
.RS 4
.RE
.PP
"The session key is unrecognized" [28]
.RS 4
.RE
.PP
"The sessionKey is expired\&. The session is closed\&." [29]
.RS 4
.RE
.PP
"The closure policy is unknown" [42]
.RS 4
.RE
.SH "AUTHOR"
.PP
\fB SysFera SAS\fR <\&info@sysfera.com\&>
.RS 4
Developed the first version of these man pages as part of the VISHNU project.
.RE
.SH "COPYRIGHT"
.br
Copyright \(co 2011 SysFera SAS
.br
.PP
These manual pages are provided under the following conditions:
.PP
Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
.sp
.RS 4
.ie n \{\
\h'-04' 1.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  1." 4.2
.\}
Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
.RE
.sp
.RS 4
.ie n \{\
\h'-04' 2.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  2." 4.2
.\}
Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
.RE
.PP
This software is governed by the CECILL licence under French law and abiding by the rules of distribution of free software. You can use, modify and/ or redistribute the software under the terms of the CeCILL license as circulated by CEA, CNRS and INRIA at the following URL "http://www.cecill.info".
.PP
As a counterpart to the access to the source code and rights to copy, modify and redistribute granted by the license, users are provided only with a limited warranty and the software's author, the holder of the economic rights, and the successive licensors have only limited liability.
.PP
In this respect, the user's attention is drawn to the risks associated with loading, using, modifying and/or developing or reproducing the software by the user in light of its specific status of free software, that may mean that it is complicated to manipulate, and that also therefore means that it is reserved for developers and experienced professionals having in-depth computer knowledge. Users are therefore encouraged to load and test the software's suitability as regards their requirements in conditions enabling the security of their systems and/or data to be ensured and, more generally, to use and operate it in the same conditions as regards security.
.sp


UMS/src/cli/listSessions.cc
#include "listSessions.hh"
#include "utils.hh"

namespace po = boost::program_options;

using namespace std;

int main (int ac, char* av[]){


		/******* Parsed value containers ****************/

		string dietConfig;

		int sessionInactivityDelay;

		std::string machineId;

		std::string userId;

		std::string sessionId;

		long startDateOption;

		long endDateOption;

		std::string sessionKey;

		/********** EMF data ************/


		UMS_Data::StatusType status;

		UMS_Data::SessionCloseType sessionClosePolicy;

		UMS_Data::ListSessions listSession;

		UMS_Data::ListSessionOptions listOptions;

/**************** Describe options *************/

		Options opt(av[0] );

        opt.add("dietConfig,c",
						            "The diet config file",
												ENV,
												dietConfig);

		opt.add("status,t",
				        "specifies the type of the sessions which will be\n"
								"listed (INACTIVE or ACTIVE)",
						  CONFIG,
						  status);

		opt.add("sessionClosePolicy,p",
				      "Specifies the closure mode of the sessions which\n"
							"will be listed (CLOSE_ON_TIMEOUT or CLOSE_ON_DISCONNECT)",
						  CONFIG,
						  sessionClosePolicy);

		opt.add("sessionInactivityDelay,d",
				      "specifies the inactivity delay in seconds\n"
							"of the sessions which will be listed",
					  CONFIG,
					  sessionInactivityDelay);

		opt.add("machineId,m",
				       "allows the user to list sessions\n"
							 "opened on a specific machine",
						CONFIG,
						machineId);

		opt.add("adminListOption,a",
				    "is an admin option for listing\n"
						"all sessions of all users",
						CONFIG);

		opt.add("userId,u",
				        "is an admin option for listing\n"
								"sessions opened by a specific user	",
						    CONFIG,
						    userId);

		opt.add("sessionId,i",
				        "allows the user to list all commands\n"
								"launched within a specific session",
						    CONFIG,
						    sessionId);

				opt.add("startDateOption,s",
									    	"allows the user to organize the commands listed\n"
												"by providing the start date\n"
												"(the UNIX timestamp	of the start date is used)",
												CONFIG,
												startDateOption);

				opt.add("endDateOption,e",
									    	"allows the user to organize the commands listed\n"
												"by providing the end date (the timestamp of the end date is used).\n"
												"By default, the end date is the current day",
												CONFIG,
												endDateOption);


				opt.add("sessionKey",
												"The session key",
												ENV,
												sessionKey);

	try {
/**************  Parse to retrieve option values  ********************/

		opt.parse_cli(ac,av);

		bool isEmpty=opt.empty();//if no value was given in the command line

		opt.parse_env(env_name_mapper());

		opt.notify();


/********  Process **************************/

		if (opt.count("status")){

			cout <<"The session state type is " << status <<endl;

			listOptions.setStatus(status);
		}

		if (opt.count("sessionClosePolicy")){

			cout <<"The session close policy is " << sessionClosePolicy<< endl;

			listOptions.setSessionClosePolicy(sessionClosePolicy);
		}

		if (opt.count("sessionInactivityDelay")){

			cout <<"The session inactivity delay is " << sessionInactivityDelay<< endl;

			listOptions.setSessionInactivityDelay(sessionInactivityDelay);
		}


		if (opt.count("machineId")){

			cout <<"The machine identifier is " << machineId << endl;

			listOptions.setMachineId(machineId);
		}

		if (opt.count("adminListOption")){

			cout <<"It is an admin list option " << endl;

			listOptions.setAdminListOption(true);
		}


		if (opt.count("userId")){

			cout <<"The user identifier is " << userId << endl;

			listOptions.setUserId(userId);

		}

		if (opt.count("sessionId")){

			cout <<"The session identifier is " << sessionId << endl;

			listOptions.setSessionId(sessionId);

		}

		if(opt.count("startDateOption")){

			cout <<"the start date option is : " << startDateOption << endl;

			listOptions.setStartDateOption(startDateOption );
		}


		if(opt.count("endDateOption")){

			cout <<"the end date option is : " << endDateOption << endl;

			listOptions.setEndDateOption(endDateOption );
		}


		if(opt.count("sessionKey")){

			cout <<"the session key is : " << sessionKey << endl;
		}


		checkVishnuConfig(opt);

		if ( opt.count("help")){

			cout << "Usage: " << av[0] <<" [options]  "<<endl;

			cout << opt << endl;

			return 0;
		}




/************** Call UMS connect service *******************************/


               // initializing DIET

							  if (diet_initialize(dietConfig.c_str(), ac, av)) {
                    cerr << "DIET initialization failed !" << endl;
               return 1;
              }



							 listSessions(sessionKey,listSession,listOptions);


							// Display the list
      if(isEmpty) {
          cout << listSession << std::endl;
      }
      else {
            for(int i = 0; i < listSession.getSessions().size(); i++) {
              cout << listSession.getSessions().get(i) << endl;
            }
       }

	}// End of try bloc

	catch(std::exception& e){
		cout << e.what() <<endl;
		return 1;
	}

	return 0;

}// end of main


